VimeoNetworkingExample-iOS/VimeoNetworkingCommonTests/Request+ProgrammedContent.swift

```diff
@@ -0,0 +1,140 @@
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import XCTest
+import OHHTTPStubs
+import VimeoNetworking
+
+class Request_ProgrammedContent: XCTestCase
+{
+    override func setUp()
+    {
+        super.setUp()
+        
+        VimeoClient.configureSharedClient(withAppConfiguration: AppConfiguration(clientIdentifier: "{CLIENT_ID}",
+                                                                          clientSecret: "{CLIENT_SECRET}",
+                                                                          scopes: [.Public, .Private, .Purchased, .Create, .Edit, .Delete, .Interact, .Upload],
+                                                                          keychainService: "com.vimeo.keychain_service",
+                                                                          apiVersion: "3.3.1"))
+    }
+    
+    override func tearDown()
+    {
+        super.tearDown()
+        
+        OHHTTPStubs.removeAllStubs()
+    }
+    
+    func test_CinemaRequest_onSuccess_returnsCorrectData()
+    {
+        let request = CinemaContentRequest.getCinemaContentRequest()
+        
+        stub(condition: isPath("/programmed/cinema")) { _ in
+            // Stub it with our "cinema-success-response.json" stub file (which is in same bundle as self)
+            let stubPath = OHPathForFile("cinema-success-response.json", type(of: self))
+            return fixture(filePath: stubPath!, headers: ["Content-Type":"application/json"])
+        }
+        
+        let expectation = self.expectation(description: "Network call expectation")
+        
+        _ = VimeoClient.sharedClient.request(request) { response in
+            switch response
+            {
+            case .success(let result):
+                XCTAssertNotNil(result.model)
+                
+                let cinemaContent = result.model
+                XCTAssertEqual(cinemaContent.count, 6)
+
+                let cinemaNames = ["John Early's Picks", "Action Sports", "Eye Candy", "Documentary", "Comedy", "The Refugee Crisis"]
+                for i in 0 ..< cinemaNames.count
+                {
+                    XCTAssertEqual(cinemaNames[i], cinemaContent[i].name)
+                    XCTAssertEqual(cinemaContent[i].content?.count, 5)
+                }
+                
+            case .failure(let error):
+                XCTFail("\(error)")
+            }
+            
+            expectation.fulfill()
+        }
+        
+        self.waitForExpectations(timeout: 1.0) { error in
+            if let unWrappedError = error
+            {
+                XCTFail("\(unWrappedError)")
+            }
+        }
+    }
+    
+    func test_CinemaRequest_onFailure_returnsError()
+    {
+        let request = CinemaContentRequest.getCinemaContentRequest()
+        
+        stub(condition: isPath("/programmed/cinema")) { _ in
+            
+            let fakeErrorJSON = ["link": NSNull(),
```

Comment:
It'll be nice if we have a file whose content is this JSON blob, then just like in the first test case, we can just read off from that file. In addition, we can refactor out the code that reads from JSON files into a function.